app/models/vm_migrate_workflow.rb

```diff
@@ -36,6 +39,14 @@ def get_source_and_targets(refresh = false)
     else
       add_target(:placement_dc_name, :datacenter, EmsFolder, result)
     end
+
+    unless field_supported(:cluster)
+      # If the user can not pick a cluster there can only be one to select
+      # from => preselect it so the hosts will be filtered accordingly.
+      cluster = vms.map(&:ems_cluster).first
```

Comment:
this way you load `ems_cluster` for every vm. Cant you just use the cluster of the first?